src/Datadog.Trace/HttpOverStreams/DatadogHttpHeaderHelper.cs

```diff
@@ -0,0 +1,33 @@
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.HttpOverStreams
+{
+    internal class DatadogHttpHeaderHelper
+    {
+        public const string CrLf = "\r\n";
+        public static readonly int CrLfLength = CrLf.Length;
+
+        public static async Task WriteLeadingHeaders(HttpRequest request, StreamWriter writer)
+        {
+            // optimization opportunity: cache the ascii-encoded bytes of commonly-used headers
```

Comment:
I'll be looking to cache these bytes and write all at once after I start getting measurements in the reliability env.

Reply:
I had cached these in a buffer to write, but started getting bad request errors, will do so in a follow up.